f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml

```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update table="sys_script_include">
+    <sys_script_include action="INSERT_OR_UPDATE">
+        <access>package_private</access>
+        <active>true</active>
+        <api_name>x_g_dis_atat.MarketResearchReport</api_name>
+        <caller_access/>
+        <client_callable>false</client_callable>
+        <description/>
+        <name>MarketResearchReport</name>
+        <script><![CDATA[var MarketResearchReport = Class.create();
+MarketResearchReport.prototype = {
+
+    errUtil: new ErrorHandler(),
+    acquisitionPackage: null,
+    janda: null,
+    fairOpportunity: new GlideRecord("x_g_dis_atat_fair_opportunity"),
+    DATA_INCOMPLETE: 4001,
+    UNEXPECTED_ERROR: 5001,
+
+    initialize: function(acqPackage) {
+        this.acquisitionPackage = acqPackage;
+        this.janda = new JustificationAndApproval(acqPackage);
+        this.constructError = this.janda.constructError;
+        this.explicitNull = this.janda.explicitNull;
+
+        this.requirementsChecklist = new RequirementsChecklist(acqPackage);
+
+        // Look up Fair Opportunity record
+        this.fairOpportunity.get(this.acquisitionPackage.fair_opportunity);
+        if (!this.fairOpportunity.sys_id) {
+            throw this.constructError(this.DATA_INCOMPLETE, "No Fair Opportunity record found");
+        }
+
+    },
+
+    explicitNull: function(value) {
+        return value ? value : null;
+    },
+
+    /**
+     * Gather the research techniques used by researchers.
+     * @param {string} techniqueSysIds CSV list of technique sys_ids defined in the reference table
+     * @return {array} technique objects with technique_value/sequence properties
+     */
+    getTechniquesUsed: function(techniqueSysIds) {
+        const PROPERTY_NAMES = ['technique_value', 'sequence'];
+        const REFERENCE_TABLE = 'x_g_dis_atat_dapps_market_research_other_techniques';
+
+        if (!techniqueSysIds) {
+            return [];
+        }
+        try {
+            return getListRecords(techniqueSysIds, PROPERTY_NAMES, REFERENCE_TABLE)
+                .filter(Dow.prototype.atatUtil.recordNotNull)
+                .sort((a, b) => { // OTHER must present last; has sequence 99
+                    return a.sequence - b.sequence;
+                })
+                .map(removeSysId);
+
+        } catch (error) {
+            throw this.errUtil.createError(
```

Comment:
I am open to being proven wrong, but I have not seen any value to the catch/re-throw pattern that prevails in the code base, unless we are actually doing something useful in the throw like translating internal errors to HTTP response codes.

Reply:
Removed try-catches from both helper methods in https://github.com/dod-ccpo/atat-snow/pull/392/commits/3d7f5576f56fb68d072c873b92739a0f778e2a51

Reply:
The problem is that you lose the context of the original error, which includes the line number where it occurred. By letting them bubble up, the system logs should have that and actually make it easier to track down. What we see in other files is a `try` followed by 100+ lines of code, so not knowing the specific line makes it far less useful in those cases. Obviously less of an issue here but re-throwing with no additional information is a pretty well known anti-pattern so I'd prefer avoiding it unless there are real benefits.